vtex-apps/return-app
# Hot fix/impersonated email

## What is this fixing?

Call center users impersonating store users cannot create a RMA. This is due to when impersonating, the app doesn't set the userId and email to the state controlling the return form. This PR fixes that, getting the profile info from sessions and using that to set those values to state, allowing a RMA to be created. Related to [this problem](https://vtex.slack.com/archives/C01D45YA3TJ/p1647513135415409).

## How to test it?

In this [workspace](https://fiximpersonate--pcbox.myvtex.com), log in as a call center operator. Navigate to the return page and create a new RMA. You should see the user email set in the profile section.

![Recording 2022-03-17 at 16 18 20](https://user-images.githubusercontent.com/38737958/158837452-63c39459-d356-48b8-978c-064ca3012d70.gif)
